Mathlib/Analysis/SpecialFunctions/ContinuousFunctionalCalculus/Rpow/Basic.lean

```diff
@@ -268,6 +269,13 @@ lemma sqrt_eq_iff (a b : A) (ha : 0 ≤ a := by cfc_tac) (hb : 0 ≤ b := by cfc
 lemma sqrt_eq_zero_iff (a : A) (ha : 0 ≤ a := by cfc_tac) : sqrt a = 0 ↔ a = 0 := by
   rw [sqrt_eq_iff a _, mul_zero, eq_comm]
 
+lemma sqrt_eq_real_sqrt (a : A) (ha : 0 ≤ a := by cfc_tac) :
```

Comment:
What's it equal to for negative `a`?

Reply:
If `¬ (0 ≤ a)` then `CFC.sqrt a = 0`. This is because `CFC.sqrt a := cfcₙ NNReal.sqrt a`, and `cfcₙ f a = 0` whenever `a` doesn't satisfy the predicate for corresponding scalar ring (in the case of `ℝ≥0`, the predicate is `(0 ≤ ·)`).

Users can prove this with `cfcₙ_apply_of_not_predicate`.

Reply:
Doesn't that mean that this theorem can lose this `ha` assumption?

Reply:
No, if `a` is selfadjoint but not nonnegative, then the LHS will be zero but the RHS will be a nonnegative element `b` such that `b ^ 2 = a⁺`.

Reply:
bors d+

Consider adding a comment to this effect to this lemma ("Note that if `ha` does not hold, while the LHS is zero, the RHS maybe not be, for example ..."), assuming it isn't very obvious to someone more familiar with the cfc.